File: vigil-frontend/src/types/review.ts
import type { PaginatedResponse } from './api';

export type ReviewStatus = 'DRAFT' | 'READY' | 'PUBLISHED' | 'PUBLISH_FAILED' | string;

export interface ReviewRead {
  id: string; // uuid
  analysis_id: string; // uuid
  status: ReviewStatus;
  summary: string | null;
  review_body: string | null;
  github_review_id: number | null;
  published_at: string | null;
  created_at: string;
  updated_at: string;
}

export type ReviewListResponse = PaginatedResponse<ReviewRead>;

File: vigil-backend/app/schemas/review.py
from datetime import datetime
from typing import List, Optional
import uuid
from pydantic import BaseModel, ConfigDict
from app.schemas.base import PaginatedResponse


class ReviewRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    analysis_id: uuid.UUID
    status: str
    summary: Optional[str] = None
    review_body: Optional[str] = None
    github_review_id: Optional[int] = None
    github_review_url: Optional[str] = None
    published_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime


class ReviewListResponse(PaginatedResponse):
    items: List[ReviewRead]
